src/cassette/TsxImage.hh

```diff
@@ -0,0 +1,133 @@
+/*
+	Copyleft 2017
+	NataliaPC aka @ishwin74
+*/
+#ifndef TSXIMAGE_HH
+#define TSXIMAGE_HH
+
+#include "CassetteImage.hh"
+#include "openmsx.hh"
+#include "endian.hh"
+#include <vector>
+
+namespace openmsx {
+
+class CliComm;
+class Filename;
+class FilePool;
+
+
+/**
+ * Code based on "CasImage" class
+ */
+class TsxImage final : public CassetteImage
+{
+public:
+	TsxImage(const Filename& fileName, FilePool& filePool, CliComm& cliComm);
+
+	// CassetteImage
+	int16_t getSampleAt(EmuTime::param time) override;
+	EmuTime getEndTime() const override;
+	unsigned getFrequency() const override;
+	void fillBuffer(unsigned pos, int** bufs, unsigned num) const override;
+
+private:
+#pragma pack(push, 1)
+	struct Block10 {
+		uint8_t     id;
+		Endian::L16 pausems;		//Pause after this block in milliseconds
+		Endian::L16 len;			//Length of data that follow
+		byte        data[0];		//Data as in .TAP files
+	};
+	struct Block11 {
+		uint8_t     id;
+		Endian::L16 pilot;			//Length of PILOT pulse {2168}
+		Endian::L16 sync1;			//Length of SYNC first pulse {667}
+		Endian::L16 sync2;			//Length of SYNC second pulse {735}
+		Endian::L16 zero;			//Length of ZERO bit pulse {855}
+		Endian::L16 one;			//Length of ONE bit pulse {1710}
+		Endian::L16 pilotlen;		//Length of PILOT tone (number of pulses) {8063 header (flag<128), 3223 data (flag>=128)}
+		uint8_t     lastbyte;		//Used bits in the last byte (other bits should be 0) {8}
+		Endian::L16 pausems;		//Pause after this block (ms.) {1000}
+		Endian::L24 len;			//Length of data that follow
+		byte	    data[0];			//Data as in .TAP files
```

Comment:
The comments have a different alignment here in Github. That's likely because we are mixing spaces and tabs.